### DISCUSSION TITLE
MQTT - multiple BMSes

### DISCUSSION THREAD

**Discussion by nedhead8233** [Louisvdw/dbus-serialbattery#145] [General]
Hello,
  I've recently added a rPi4 running Venus OS to my solar setup.  I have two batteries that have the overkill solar BMSes.  I found this driver and am thrilled.  Both batteries show up in the vrm portal and show the correct details in the remote console.  Part of the reason that I'm adding this advanced monitoring to my system is that I had an issue with one of my BMSes; the cell over voltage protection was not working properly.  On occasion, I would see a cell at 3.8 or 3.9v when the cut-off was set at 3.65.  I'd like to automate writing both batteries individual cell voltages to a database.  With MQTT enabled, using MQTT Explorer, I can only see one battery.  I believe what is happening is that both batteries are publishing data on the same topic.  They are both showing up in VRM as  "SerialBattery (LLT/JBD) v0.12".  In the MQTT Explorer, I don't see any unique identifier for the one battery that does show up.  If I watch the data in the Voltages topic update, it seems to flip back and forth between two values.  I'm wondering if there is a way to set a unique name for each battery, so that it will publish on it's own topic.   Or perhaps, I'm off and the issue is something else.  Have any of you run into this or gotten multiple Batteries to publish MQTT?    Below is a link to a screenshot of my topic tree.  [https://drive.google.com/file/d/1sVjjoDN4AfiumXxl7VElnN6cQqwKC1Yr/view?usp=sharing](https://drive.google.com/file/d/1sVjjoDN4AfiumXxl7VElnN6cQqwKC1Yr/view?usp=sharing)

**Comment by Louisvdw** [comment]
The issue is because the driver does not officially support mutiple battery banks. When we have that workking it should work for you to see both on MQTT as well. Follow the #8 ticket to get notifications

**Reply by SergiCEES** [reply]
If I understood well, once the driver is ready to manage parallel banks as stated in ticket #8 , the driver is going to be making some kind of average and showing the final results (not only the principal). But, is the driver going to be able to publish each battery state to MQTT? 

Without knowing the internals of the driver, from my pov it shoud make some kind of average to show to Venus OS as 1 battery pack (composed by different parallel batteries) and from them the Venus OS is going to determine the DVCC and limit (or not) the charge/discharge, but from mqtt it's going to be awesome if every pack could be shown as battery 0 (average of all #8) but then battery 1, battery 2 and so on and so on. @nedhead8233 @Louisvdw

**Reply by herrep** [reply]
I also understand ticket [#8 ](https://github.com/Louisvdw/dbus-serialbattery/issues/8) in the sense that access to combined battery modules is given so that it remains unclear whether the details of each single battery module can be published via MQTT when the support for multiple battery modules is available.

**Comment by Louisvdw** [comment]
A lot depend on the final implementation for this feature, so I cannot give you all the answers now. We will have to see what can be done.
The driver itself does not publish anything on MQTT. That is a feature of the GX system which publish what is available on the DBUS, but I don't know if it filters and only publish spesific DBUS paths or just blindly everything (I would guess only some paths).

The initial implentation will closely follow what is implemented for multi bank modules for other batteries on the paths that Victron use. So that will give the same values as other batteries have.

**Comment by mr-manuel** [comment]
Did you try with the Venus OS >v2.92? Every battery should be published to MQTT with this driver.
